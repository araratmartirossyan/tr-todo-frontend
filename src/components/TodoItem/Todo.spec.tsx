import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { TodoItem, Props } from './index'

interface TestState extends Props {
  title: string
  done: boolean
  _id: string
  onRemove: jest.Mock
  onUpdate: jest.Mock
}

describe('TodoItem test', () => {
  const defaultProps: TestState = {
    title: 'Test Task',
    done: false,
    _id: '11',
    onRemove: jest.fn(),
    onUpdate: jest.fn(),
  }

  const createComponent = (testProps: TestState) =>
    shallow(<TodoItem {...testProps} />)

  it('should render component', () => {
    const component = createComponent(defaultProps)
    expect(toJson(component)).toMatchSnapshot()
  })

  it('on edit button click Edit mode Component should appear', () => {
    const component = createComponent(defaultProps)
    const editBtn = component.find('[data-test-id="todo-edit-button"]')
    expect(editBtn.exists()).toBe(true)
    editBtn.simulate('click', {
      stopPropagation: () => {},
    })

    const editComponent = component.find('[data-test-id="todo-edit-task"]')
    expect(editComponent.exists()).toBe(true)
  })

  it('on remove button click it should fire onRemove event', () => {
    const component = createComponent(defaultProps)
    const editBtn = component.find('[data-test-id="todo-remove-button"]')
    expect(editBtn.exists()).toBe(true)
    editBtn.simulate('click', {
      stopPropagation: () => {},
    })

    expect(defaultProps.onRemove).toBeCalled()
  })

  it('on toggle checkbox click it should fire onUpdate event', () => {
    const component = createComponent(defaultProps)
    const editBtn = component.find('[data-test-id="todo-checkbox-toogle"]')
    expect(editBtn.exists()).toBe(true)
    editBtn.simulate('change')

    expect(defaultProps.onUpdate).toBeCalledWith({
      title: 'Test Task',
      done: true,
      _id: '11',
    })
  })
})
