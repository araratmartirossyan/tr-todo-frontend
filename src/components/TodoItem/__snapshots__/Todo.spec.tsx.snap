// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoItem test should render component 1`] = `
<styled.li
  done={false}
  onClick={[Function]}
>
  <Checkbox
    checked={false}
    className="todo-checkbox"
    data-test-id="todo-checkbox-toogle"
    name="done"
    onChange={[Function]}
  />
  <span>
    Test Task
  </span>
  <styled.div>
    <Button
      className="todo-icon-button"
      data-test-id="todo-edit-button"
      onClick={[Function]}
      tertiary={true}
    >
      <Component />
    </Button>
    <Button
      className="todo-icon-button"
      data-test-id="todo-remove-button"
      onClick={[Function]}
      tertiary={true}
    >
      <Component />
    </Button>
  </styled.div>
</styled.li>
`;
